File: src/com/method/demo_2364.java
package com.method;

public class demo_2364 {
}


class Solution {
    public long countBadPairs(int[] nums) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        long res = 0;
        for (int i = 0; i < nums.length; i++) {
            int key = nums[i] - i;
            res += i - mp.getOrDefault(key, 0);
            mp.put(key, mp.getOrDefault(key, 0) + 1);
        }
        return res;
    }
}

File: src/solutions/demo_2364.js
const countBadPairs = (nums)=>{
    const mp = new Map();
    let res = 0;
    for (let i = 0;i <nums.length;i++){
        const key = nums[i] - i;
        res += i - (mp.get(key) || 0);
        mp.set(key,(mp.get(key)||0) +1);
    }
    return res;
}